pkg/frame/compressed.go

```diff
@@ -2,11 +2,89 @@ package frame
 
 import (
 	"bytes"
+	"errors"
 	"image"
 	"image/jpeg"
 )
 
+/* addMotionDht, dhtMarker, dht, and sosMarker is protected under the following license:
+
+The MIT License (MIT)
+
+Copyright (c) 2016 Philip Thomas Casado
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+*/
```

Comment:
@at-wat like this?

Reply:
Maybe better to move the copied part to separated file (if code inside `decodeMJPEG` and `addMotionDht` is also copied, current structure is fine)
Could you add a URL of the original repo as well?

Other things looks good to me!

Reply:
Link is added! parts of the functions you mentioned are taken from the repo